packages/blocks/src/root-block/widgets/element-toolbar/align-button.ts

```diff
@@ -232,24 +324,88 @@ export class EdgelessAlignButton extends WithDisposable(LitElement) {
     });
   }
 
+  private _splitElementsToChunks(models: GfxModel[]) {
+    const sortByCenterX = (a: GfxModel, b: GfxModel) =>
+      a.elementBound.center[0] - b.elementBound.center[0];
+    const sortByCenterY = (a: GfxModel, b: GfxModel) =>
+      a.elementBound.center[1] - b.elementBound.center[1];
+    const elements = models.filter(ele => {
+      if (
+        ele instanceof ConnectorElementModel &&
+        (ele.target.id || ele.source.id)
+      ) {
+        return false;
+      }
+      return true;
+    });
+    elements.sort(sortByCenterY);
+    const chunks = chunk(elements, 4);
+    chunks.forEach(items => items.sort(sortByCenterX));
+    return chunks;
+  }
+
+  private _updatChildElementsXYWH(
+    container: GfxContainerElement,
+    targetBound: Bound
+  ) {
+    const containerBound = Bound.deserialize(container.xywh);
+    const scaleX = targetBound.w / containerBound.w;
+    const scaleY = targetBound.h / containerBound.h;
+    container.childElements.forEach(child => {
+      const childBound = Bound.deserialize(child.xywh);
+      childBound.x = targetBound.x + scaleX * (childBound.x - containerBound.x);
+      childBound.y = targetBound.y + scaleY * (childBound.y - containerBound.y);
+      childBound.w = scaleX * childBound.w;
+      childBound.h = scaleY * childBound.h;
+      this._updateXYWH(child, childBound);
+    });
+  }
+
   private _updateXYWH(ele: BlockSuite.EdgelessModel, bound: Bound) {
     if (ele instanceof ConnectorElementModel) {
       ele.moveTo(bound);
-    } else if (ele instanceof GroupElementModel) {
-      const groupBound = Bound.deserialize(ele.xywh);
-      ele.childElements.forEach(child => {
-        const newBound = Bound.deserialize(child.xywh);
-        newBound.x += bound.x - groupBound.x;
-        newBound.y += bound.y - groupBound.y;
-        this._updateXYWH(child, newBound);
+    } else if (isGfxContainerElm(ele)) {
+      this._updatChildElementsXYWH(ele, bound);
+      this.edgeless.service.updateElement(ele.id, {
+        xywh: bound.serialize(),
       });
+    } else if (ele instanceof LayoutableMindmapElementModel) {
```

Comment:
Since the `LayoutableMindmapElelementModel` is also a `GfxContainerElement`, I think this branch is unreachable.

Reply:
Thanks!